vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -296,6 +296,39 @@ testFips() {
   echo "$test:Finish"
 }
 
+testCloudInit() {
+  test="testCloudInit"
+  echo "$test:Start"
+  os_sku=$1
+
+  # Limit this test only to Mariner or Azurelinux 
+  if [[ "$os_sku" == "CBLMariner" || "$os_sku" == "AzureLinux" ]]; then
+    FILE=/var/log/cloud-init.log
+    if test -f "$FILE"; then
+      echo "Check cloud-init log exist."
+      if grep 'WARNING\|ERROR' $FILE; then
+        msg=$(grep 'WARNING\|ERROR' $FILE)
+        err $test "Cloud-init log has WARNING/ERROR: ${msg}."
```

Comment:
I'm not a fan of running this command twice. Consider running it once and then checking the status (note: I haven't run this so you'll need to verify it):

```suggestion
      local grep_output=$(grep 'WARNING\|ERROR' ${FILE})
      local grep_status=$?
      if [ ${grep_status} -eq 0 ]; then
        err $test "Cloud-init log has WARNING/ERROR: ${grep_output}."
```